rad/prelude/test.rad

```diff
@@ -0,0 +1,60 @@
+;; testing functions
+
+(def is-test-env
+  "True iff file is being run as part of the Haskell suite"
+  (catch 'any (eq? test-env__ #t) (fn [x] #f)))
+
+;; Annoyingly verbose since we have neither the prelude nor pattern-matching.
+(def eval
+  (fn [expr env]
+    (if (list? expr)
+        (if (> (foldr (fn [x acc] (+ acc 1)) 0 expr) 0)
```

Comment:
Considered your `(eq? expr nil)` idea, but since there isn't even yet a `not`, it'd involve changing the order of the consequents and that seemed a little ugly

Reply:
Can't tell if joke.

Reply:
Nope :P